frontend/ui/widget/datetimewidget.lua

```diff
@@ -84,6 +84,34 @@ local DateTimeWidget = FocusManager:extend{
     extra_callback = nil,
 }
 
+local IMPOSSIBLE_DATES = {
+    -- 29th February is handled separately
+    { month = 2, day = 30 },
+    { month = 2, day = 31 },
+    -- April, June, September, November have 30 days
+    { month = 4, day = 31 },
+    { month = 6, day = 31 },
+    { month = 9, day = 31 },
+    { month = 11, day = 31 },
+}
+
+function DateTimeWidget:isValidDate(year, month, day)
```

Comment:
Reinventing the wheel.
https://github.com/koreader/koreader/blob/a15c294bc2f645ba65f1fc4de6e2da537c11ef6a/frontend/ui/widget/numberpickerwidget.lua#L322